ledger/src/ui/test/bootloader/test_bootloader.c

```diff
@@ -0,0 +1,575 @@
+static bool G_authorize_signer_called = false;
+static bool G_get_retries_called = false;
+static bool G_unlock_called = false;
+static unsigned char G_autoexec = 0;
+static bool G_reset_attestation_called = false;
+static bool G_reset_signer_authorization_called = false;
+static bool G_reset_onboard_called = false;
+
+#define RESET_IF_STARTED_CALLED()                                         \
+    (G_reset_attestation_called && G_reset_signer_authorization_called && \
+     G_reset_onboard_called)
+
+// Helper functions
+static void reset_flags() {
+    G_host_seed_is_set = false;
+    G_pin_buffer_updated = false;
+    G_is_onboarded = false;
+    G_is_pin_set = false;
+    G_is_pin_buffer_cleared = false;
+    G_get_attestation_called = false;
+    G_authorize_signer_called = false;
+    G_get_retries_called = false;
+    G_unlock_called = false;
+    G_autoexec = 0;
+    G_reset_attestation_called = false;
+    G_reset_signer_authorization_called = false;
+    G_reset_onboard_called = false;
+}
+
+// Mock function calls
+unsigned int set_host_seed(volatile unsigned int rx, onboard_t* onboard_ctx) {
+    ASSERT_NOT_NULL(onboard_ctx);
+    G_host_seed_is_set = true;
+    return 0;
+}
+
+unsigned int update_pin_buffer(volatile unsigned int rx) {
+    G_pin_buffer_updated = true;
+    return 3;
+}
+
+unsigned int is_onboarded() {
+    SET_APDU_AT(1, G_is_onboarded);
+    SET_APDU_AT(2, VERSION_MAJOR);
+    SET_APDU_AT(3, VERSION_MINOR);
+    SET_APDU_AT(4, VERSION_PATCH);
+    return 5;
+}
+
+unsigned int onboard_device(onboard_t* onboard_ctx) {
+    G_is_onboarded = true;
+    return 3;
+}
+
+void clear_pin() {
+    G_is_pin_buffer_cleared = true;
+}
+
+unsigned int set_pin() {
+    G_is_pin_set = true;
+    return 3;
+}
+
+unsigned int echo(unsigned int rx) {
+    return rx;
+}
+
+unsigned int get_mode() {
+    SET_APDU_AT(1, RSK_MODE_BOOTLOADER);
+    return 2;
+}
+
+unsigned int get_attestation(volatile unsigned int rx, att_t* att_ctx) {
+    ASSERT_NOT_NULL(att_ctx);
+    G_get_attestation_called = true;
+    return 3;
+}
+
+unsigned int do_authorize_signer(volatile unsigned int rx,
+                                 sigaut_t* sigaut_ctx) {
+    ASSERT_NOT_NULL(sigaut_ctx);
+    G_authorize_signer_called = true;
+    return 3;
+}
+
+unsigned int get_retries() {
+    G_get_retries_called = true;
+    return 3;
+}
+
+unsigned int unlock() {
+    G_unlock_called = true;
+    return 3;
+}
+
+void set_autoexec(char value) {
+    G_autoexec = value;
+}
+
+void reset_attestation(att_t* att_ctx) {
+    G_reset_attestation_called = true;
+}
+
+void reset_signer_authorization(sigaut_t* sigaut_ctx) {
+    G_reset_signer_authorization_called = true;
+}
+
+void reset_onboard_ctx(onboard_t* onboard_ctx) {
+    G_reset_onboard_called = true;
+}
+
+// Function definitions required for compiling bootloader.c
+void init_signer_authorization() {
+}
+
+unsigned short io_exchange(unsigned char channel, unsigned short tx_len) {
+    return 0;
```

Comment:
Since we're here, we could check this is called with the correct parameters (namely, the correct `channel`).